src/components/CommentForm.jsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from "react";
+import { postComment } from "../utils/api";
+
+// export default function CommentForm({ article_id, onCommentSubmit }) {
+//   const [commentBody, setCommentBody] = useState("");
+//   const [error, setError] = useState(null);
+
+//   const handleSubmit = (e) => {
+//     e.preventDefault();
+//     const newComment = { body: commentBody, author: "grumpy19" }; //replace with dynamic username
+//     postComment(article_id, newComment)
+//       .then((comment) => {
+//         onCommentSubmit(comment);
+//         setCommentBody("");
+//         setError(null);
+//       })
+//       .catch((error) => {
+//         console.error("Error posing comment", error);
+//         setError("Failed to post comment. PLease try again");
+//       });
+//   };
+
+//   return (
+//     <form onSubmit={handleSubmit}>
+//       <textarea
+//         value={commentBody}
+//         onChange={(e) => setCommentBody(e.target.value)}
+//         placeholder="Speak your mind...."
+//       />
+//       <button type="submit">Post Comment</button>
+//       {error && <p className="error">{error}</p>}
+//     </form>
+//   );
+// }
+
+export default function CommentForm({ article_id, onCommentSubmit }) {
+  // Accept articleId as a prop
+  const initialCommentState = {
+    votes: "",
+    created_at: "",
+    author: "grumpy19",
+    body: "",
+    article_id: article_id, // Initialize with the passed articleId
+  };
```

Comment:
The body doesn't require votes, created_at and article_id on backend. Only needs author and body as article_id is on the url and votes and created_at have default value